src/test/forms/formHelpers.test.js

```diff
@@ -0,0 +1,48 @@
+import React from 'react';
+import { shallow } from 'enzyme';
+
+import { ManagedNumberField } from '../../forms/formHelpers';
+
+test('ManagedNumberField should match snapshot', () => {
+  const wrapper = shallow(
+    <ManagedNumberField
+      format={ jest.fn() }
+      value={ 0 } />
+  );
+  expect(wrapper).toMatchSnapshot();
+});
+
+test('should set focused state to true on focus', () => {
+  const value = 10;
+  const wrapper = shallow(
+    <ManagedNumberField
+      format={ jest.fn() }
+      value={ value } />
+  );
+  wrapper.find('FormInput').simulate('focus', { target: { value }}, undefined);
+  expect(wrapper.state('focused')).toBe(true);
+});
+
+test('should blank focusedVal state before focus if current value of FormInput is 0', () => {
+  const inputValue = 0;
+  const wrapper = shallow(
+    <ManagedNumberField
+      format={ jest.fn() }
+      value={ 0 } />
+  );
+  wrapper.find('FormInput').simulate('focus', { target: { value: inputValue }}, undefined);
+  expect(wrapper.state('focusedVal')).toBe('');
+});
+
+test('should set focused state to false on blur', () => {
+  const value = 10;
+  const wrapper = shallow(
+    <ManagedNumberField
+      format={ jest.fn() }
+      onBlur={ jest.fn() }
+      value={ value } />
+  );
+  wrapper.setState({ focused: true });
```

Comment:
Might want to simulate focus here rather than explicitly setting `focused` to true, to mimic exactly how it would work in a real interaction.

Reply:
I chose to explicitly set focused=true in the onBlur test to keep this test specific to testing the onBlur call, and let the onFocus test take care of simulating focus. (I don't have an explicit focused=false statement in the onFocus test because the default value of focused is false anyway.) I would suggest keeping it this way to keep the tests specific.

Reply:
I would argue that, without assuming anything about how `handleFocus` works, setting `focused` to true explicitly could put the component in an invalid state. Although maybe that doesn't matter for such a simple test, and doing it this way does mean the test will still work even if `handleFocus` is broken. 

Although now that I think of it, shouldn't we be testing the props of the underlying `input` component, not the state of our component which could change arbitrarily?

Reply:
I thought about whether the FormInput component itself needed to be tested. Looking at the component, it appears to be an out-of-the-box semantic-ui-react component. If you take a look at the snapshot file, there's very little that could be tested inside of FormInput -- we could replace the function props with spies to make sure that onBlur, onFocus, etc. are called, but that would suggest we're not trusting the semantic-ui-react module to have correctly wired onBlur/onFocus/etc. Only by rendering the test at the ManagedNumberField level can we see that FormInput's onBlur calls the ManagedNumberField's handleBlur function.

Running the state-setting unit tests at this level is also consistent with how it was done in the React/Jest tutorial I took, if that helps.

I think your concern is addressed by having the snapshot test. If the ManagedNumberField component is modified to the extent that it renders differently, the snapshot test will catch that. If the ManagedNumberField component is modified so that a function passed down as a prop changes (e.g. it passes down a different function to FormInput as the onBlur prop), the render doesn't change so the snapshot test would pass, but the corresponding state-setting test in ManagedNumberField would fail.

For your reference: I went through a lengthy React tutorial that included testing with Jest/Enzyme, but this is the first time I'm writing tests in a non-tutorial environment -- which is why I started with a few small unit tests :-) So I'm basing my comments on the principles I learned in the tutorial.

Reply:
OK, I think you're right about not testing `Form.Input` in this case actually. I was thinking that the focused state of the underlying input was managed by `ManagedNumberField` similarly to its `value` (i.e. passed into `Form.Input` as a prop), but I now realize that the `focused` field just keeps track of the focused state of the component, which is managed by `Form.Input`. When we get to making tests about the value of the input, we should check that the `Form.Input` value prop matches what we want it to be.

The other issue I had is pretty minor the more I think about it, so I think this is ready to be merged.